src/main/scala/temple/generate/target/openapi/OpenAPIBuilder.scala

```diff
@@ -0,0 +1,129 @@
+    case FloatType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "double" else "float", Seq(minimum, maximum).flatten: _*)
+    case ForeignKey(references) =>
+      OpenAPISimpleType("number", "int32", "description" -> s"Reference to $references ID".asJson)
+  }
+
+  private def generateItemType(attributes: Map[String, Attribute]): OpenAPIObject = OpenAPIObject(
+    attributes.iterator
+      .filter { case _ -> attribute => !isServerAttribute(attribute) }
+      .map { case str -> attribute => str -> attributeToOpenAPIType(attribute) }
+      .to(attributes.mapFactory),
+  )
+
+  private def jsonContent(mediaTypeObject: MediaTypeObject) = Map("application/json" -> mediaTypeObject)
+
+  def addPaths(service: Service): this.type = {
+    val lowerName       = service.name.toLowerCase
+    val capitalizedName = service.name.capitalize
+    val tags            = Seq(capitalizedName)
+    service.endpoints.foreach {
+      case ReadAll =>
+        path(s"/$lowerName/all") += HTTPVerb.Get -> Path(
+            s"Get a list of every $lowerName",
+            tags = tags,
+            responses = Map(
+              200 -> Literal(
+                jsonContent(MediaTypeObject(OpenAPIArray(generateItemType(service.attributes)))),
+                s"$capitalizedName list successfully fetched",
+              ),
+              500 -> Response.Ref(useError(500)),
+            ),
+          )
+      case Create => // TODO in future PR
+      case Read   => // TODO in future PR
+      case Update => // TODO in future PR
+      case Delete => // TODO in future PR
+    }
+    this
+  }
+
+  def useError(code: Int): String = {
+    errorTracker.flag(code)
+    s"Error$code"
+  }
+
+  def errorBlock: Map[String, Response] = errorTracker.view.map { case i -> response => useError(i) -> response }.toMap
+
+  def toOpenAPI: OpenAPIFile = OpenAPIFile(
+    info = Info(name, version, description),
+    paths = paths.view.mapValues(_.toMap).toMap,
+    components = Components(responses = errorBlock),
+  )
+}
+
+object OpenAPIBuilder {
+
+  private def build(name: String, version: String, description: String = "")(services: Service*): OpenAPIFile = {
+    val builder = new OpenAPIBuilder(name, version, description)
+    services.foreach(builder.addPaths)
+    builder.toOpenAPI
+  }
+
+  def render(name: String, version: String, description: String = "")(services: Service*): String =
+    Printer(preserveOrder = true, dropNullKeys = true).pretty(build(name, version, description)(services: _*).asJson)
+
+  /** Create a Response representation for an error */
+  private[openapi] def generateError(description: String, example: String): Response =
+    Literal(
+      description = description,
+      content = ListMap(
+        "application/json" -> MediaTypeObject(
```

Comment:
Do you not want to use your `jsonContent` function here? Seems silly to have it but then not use it because it's private